hal/src/electron/rtos_hook.cpp

```diff
@@ -10,4 +28,22 @@ void vApplicationStackOverflowHook(TaskHandle_t, char*) {
 }
 #endif
 
+void vApplicationTaskDeleteHook(void *pvTaskToDelete, volatile BaseType_t* pxPendYield) {
+    (void)pvTaskToDelete;
+    (void)pxPendYield;
+
+    // Temporarily raise IDLE thread priority to (configMAX_PRIORITIES - 1) (maximum)
+    // to give it some processing time to clean up the deleted task resources.
+    vTaskPrioritySet(xTaskGetIdleTaskHandle(), configMAX_PRIORITIES - 1);
+
+    // Immediately request the scheduler to yield to now higher priority IDLE thread
+    *pxPendYield = pdTRUE;
+}
+
+void vApplicationIdleHook(void) {
+    // Restore IDLE thread priority back to the default one
+    vTaskPrioritySet(xTaskGetIdleTaskHandle(), tskIDLE_PRIORITY);
```

Comment:
How about adding a condition here, so that we only change the priority if it was raised previously?

Reply:
If you take a look at `vTaskPrioritySet` implementation, nothing will actually happen if the set priority is the same as the current one.

We'd still need to add a critical section around the check and `vTaskPrioritySet`, so I don't seem much of a performance benefit for example, it's almost 1-1.

Reply:
Since the [FreeRTOS docs](https://www.freertos.org/a00129.html) don't spell out what will happen when the same priority is set, it could change in the future for better or worse.  I agree it would be a good idea to guard against this.  Since it should only run the one time that a thread is deleted, there shouldn't be much overhead in an additional safety check.

BTW this works great on Boron!  Clever use of vApplicationTaskDeleteHook :)  This should really be a documented pattern, I see so many SO questions about how this issue.

Reply:
I'll add an atomic flag just in case :+1: